Effect.ts

```diff
@@ -0,0 +1,29 @@
+import { EffectState } from "./EffectState.ts";
+import { Context } from "./Runtime.ts";
+import { Signature } from "./Signature.ts";
+
+declare const S_: unique symbol;
+declare const V_: unique symbol;
+declare const E_: unique symbol;
+declare const T_: unique symbol;
+
+declare const effectId_: unique symbol;
+export type EffectId = Signature & { [effectId_]: true };
+
+export abstract class Effect<
+  S extends boolean,
+  V,
+  E extends Error,
+  T,
```

Comment:
It would be nice to add some names to these types

`V` is for the environment/context
`E` is for any errors
`T` is for the return type

And `S`, What is it for?

Reply:
"S" stands for "Sync" –– `true` is sync, `false` is async. Each effect is a union of its own boolean/is-sync type and those of its dependencies. At the root, we check if it should be sync with `ColoredResult`, which uses the condition `false extends S ? Promise<R> : R`. In other words, if any effect contains async computation, then the effect is NOT sync, otherwise it is.

Regarding adding names to the types, they occur frequently throughout Zones. Would rather keep it simple. However, we should definitely add some documentation + comments (#2).